lib/logic/api/helper.js

```diff
@@ -1,13 +1,26 @@
 const debug = require('debug')('codefresh:http');
 const fs = require('fs');
 const path = require('path');
-const rp = require('request-promise');
+const request = require('requestretry');
 const _ = require('lodash');
 const CFError = require('cf-errors');
-const { printError } = require('../../interface/cli/helpers/general');
+const { printError, isDebug } = require('../../interface/cli/helpers/general');
+const config = require('../../logic/cli-config/Manager').config();
 
 const { version } = JSON.parse(fs.readFileSync(path.resolve(__dirname, '../../../package.json')));
 
+const RETRY_STATUS_CODES = [502, 503, 504];
+const retryOptions = {
+    fullResponse: true,
+    maxAttempts: config.request.maxRetries,
+    retryDelay: config.request.retryDelay,
+    retryStrategy: (err, response) => !isDebug() && (request.RetryStrategies.NetworkError(err) || RETRY_STATUS_CODES.includes(response.statusCode)),
+};
+
+function _makeResponseError(response) {
```

Comment:
we already have logic that handles an error. in the catch block atm.
please use the old logic

Reply:
It's needed for BadStatusError simulation, that was thrown by previous request lib